src/message/getHtmlPieceDescriptors.js

```diff
@@ -7,17 +7,17 @@ import { isSameDay } from '../utils/date';
 export default (
   messages: $ReadOnlyArray<Message | Outbox>,
   narrow: Narrow,
-): HtmlPieceDescriptor[] => {
+): $ReadOnlyArray<HtmlPieceDescriptor> => {
   const showHeader = !isPmNarrow(narrow) && !isTopicNarrow(narrow);
 
-  let prevMessage = undefined;
-  const sections = [{ key: 0, data: [], message: null }];
-  messages.forEach(message => {
+  return messages.flatMap((message, index, arr) => {
+    const prevMessage = arr[index - 1];
```

Comment:
This makes sense but unfortunately Flow doesn't notice that it can be `undefined`.

(That's an understandable design choice on Flow's part -- it's unusual to have an array access that can actually be out of bounds like this.)

For that reason, probably best to explicitly annotate it so that Flow can check we handle the `undefined` case reasonably in the code below.

Reply:
Bump :slightly_smiling_face: